core/src/com/unciv/models/translations/TranslationFileWriter.kt

```diff
@@ -99,16 +115,25 @@ object TranslationFileWriter {
                 if (existingTranslationKeys.contains(hashMapKey)) continue // don't add it twice
                 existingTranslationKeys.add(hashMapKey)
 
-                // count translatable lines only once (e.g. for English)
-                if (language == "English") countOfTranslatableLines++
-
-                var translationValue = ""
+                // count translatable lines only once
+                if (languageIndex == 0) countOfTranslatableLines++
 
-                val translationEntry = translations[hashMapKey]
-                if (translationEntry != null && translationEntry.containsKey(language)) {
-                    translationValue = translationEntry[language]!!
-                    translationsOfThisLanguage++
-                } else stringBuilder.appendLine(" # Requires translation!")
+                var translationValue = fun (): String {
```

Comment:
This looks like it was written specifically to annoy me personally.
This is - and I don't use this lightly - barely readable.
The decision tree flow is obscured by the syntax.


Please make a function, call it a name.
Don't use .? , let, or also.
I assure you it will be much more readable.